cli/src/test/java/com/devonfw/tools/ide/context/ProcessContextGitMock.java

```diff
@@ -61,19 +60,30 @@ public void setExitCode(int exitCode) {
   }
 
   /**
-   * @return the {@link List} of mocked error messages.
+   * @return the {@link List} of mocked stderr messages.
    */
   public List<String> getErrors() {
 
-    return errors;
+    return this.outputMessages.stream().filter(OutputMessage::error).map(OutputMessage::message).collect(Collectors.toList());
```

Comment:
This will not work. Current code is calling `getErrors()` or `getOuts()` in order to add mocked messages to such `List`.
Streams will create immutable collections that cannot be modified after creation.
In case we leave it like this, also use `toList()` simplification in all places.

Could you discuss with @jan-vcapgemini who IMHO created this class about the design?
Wouldn't it be smarter to create a regular `ProcessResult` here and make it accessible so instead of tests calling
`processContextGitMock.getErrors().add("fake error message")` we could call `processContextGitMock.getProcessResult().getOutputMessages().add(new OutputMessage("fake error message", true))`
to make it work.
There is a lot of pointless redundancy in this class related to `ProcessResultImpl` that can be avoided this way.

BTW: I just noticed that you already found this problem and had to update an existing test-case.

Reply:
If you want a convenience method for adding faked out or err messages, you could still provide them here but delegating to the ProcessResult. Also this way we could even allow a setter for the mocked `ProcessResult` that also gives us more flexibility (resetting the result for a next faked git call, etc.).

Reply:
done